errors.go

```diff
@@ -125,6 +125,15 @@ func (e NotAcceptableError) Unwrap() error { return HTTPError(e) }
 // It transforms any error into the unified error type [HTTPError],
 // Using the [ErrorWithStatus] interface.
 func ErrorHandler(err error) error {
+	var errorStatus ErrorWithStatus
+	if errors.As(err, &HTTPError{}) || errors.As(err, &errorStatus) {
+		return handleHTTPError(err)
+	}
+
+	return err
+}
+
+func handleHTTPError(err error) HTTPError {
 	errResponse := HTTPError{
 		Err: err,
 	}
```

Comment:
There is something strange here

If err is already an HTTPError when entering ErrorHandler, you will created an HTTPError of an HTTPError

Reply:
I’m not sure that’s how’s errors.As works. I’m not passing the reference of the second argument for a reason. I could maybe make my intention more clear by just checking the type, but I’m fairly certain the tests wouldn’t have passed if what you say is true.

Reply:
Oh I see the concern. I’ll write a test case for it and fix the issue. Nice eye

Reply:
Wrote a test case today. It's fine actually this code covers that case

```
func handleHTTPError(err error) HTTPError {
	errResponse := HTTPError{
		Err: err,
	}

	var errorInfo HTTPError
	if errors.As(err, &errorInfo) {
		errResponse = errorInfo
	}
```